js/views/components/CurrencySelector.js

```diff
@@ -0,0 +1,144 @@
+import _ from 'underscore';
+import $ from 'jquery';
+import app from '../../app';
+import loadTemplate from '../../utils/loadTemplate';
+import { polyTFallback } from '../../utils/templateHelpers';
+import { ensureMainnetCode, isSupportedWalletCur } from '../../data/walletCurrencies';
+import baseVw from '../baseVw';
+
+export default class extends baseVw {
+  constructor(options = {}) {
+    const controlTypes = ['checkbox', 'radio'];
+
+    const controlType = options.initialState.controlType;
+    if (controlType && !controlTypes.includes(controlType)) {
+      throw new Error('If provided the controlType must be a valid value.');
+    }
+
+    if (!options.initialState.currencies ||
+      !Array.isArray(options.initialState.currencies)) {
+      throw new Error('Please provide an initial array of currencies.');
+    }
+
+    if (options.initialState.activeCurs &&
+      !Array.isArray(options.initialState.activeCurs)) {
+      throw new Error('If activeCurs are provided, they must be an array.');
+    }
+
+    const opts = {
+      ...options,
+      initialState: {
+        controlType: 'checkbox',
+        currencies: [],
+        activeCurs: [],
+        ...options.initialState,
+      },
+    };
+
+    super(opts);
+  }
+
+  get className() {
+    return 'unstyled borderStackedAll curSelector';
+  }
+
+  get tagName() {
+    return 'ul';
+  }
+
+  events() {
+    return {
+      'click .js-curControl': 'handleCurClick',
+    };
+  }
+
+  handleCurClick(e) {
+    const code = $(e.target).attr('data-code');
+    let activeCurs = [...this.getState().activeCurs];
+    const active = !activeCurs.includes(code);
```

Comment:
Because it's the new state. If the currency was inactive, and you click it, it's now active, and vice versa. Clicking it flips the value. I can rename it to something like newActiveState.